chainer_chemistry/models/gat.py

```diff
@@ -0,0 +1,208 @@
+import chainer
+import chainer.backends.cuda as cuda
+from chainer import functions
+from chainer import Variable
+
+from chainer_chemistry.config import MAX_ATOMIC_NUM
+from chainer_chemistry.links import EmbedAtomID
+from chainer_chemistry.links import GraphLinear
+
+
+class GAT(chainer.Chain):
+    """Graph Attention Networks (GAT)
+
+    See: Veličković, Petar, et al. (2017).\
+        Graph Attention Networks.\
+        `arXiv:1701.10903 <https://arxiv.org/abs/1710.10903>`_
+
+    Args:
+        out_dim (int): dimension of output feature vector
+        hidden_dim (int): dimension of feature vector
+            associated to each atom
+        n_layers (int): number of layers
+        n_atom_types (int): number of types of atoms
+        n_heads (int): number of multi-head-attentions.
+        n_edge_types (int): number of edge types.
+        drop_out_ratio (float): dropout ratio of the normalized attention
+            coefficients
+        negative_slope (float): LeakyRELU angle of the negative slope
+        concat_hidden (bool): If set to True, readout is executed in each layer
+            and the result is concatenated
+        concat_heads (bool) : Whether to concat or average multi-head
+            attentions
+        weight_tying (bool): enable weight_tying or not
+
+    """
+
+    def __init__(self, out_dim, hidden_dim=16, n_heads=8, negative_slope=0.2,
+                 n_edge_types=4, n_layers=4, dropout_ratio=-1.,
+                 n_atom_types=MAX_ATOMIC_NUM, concat_hidden=False,
+                 concat_heads=False, weight_tying=True):
+        super(GAT, self).__init__()
+        n_readout_layer = n_layers if concat_hidden else 1
+        n_message_layer = n_layers
+        with self.init_scope():
+            # Update
+            self.embed = EmbedAtomID(out_size=hidden_dim, in_size=n_atom_types)
+            self.message_layers = chainer.ChainList(
+                *[GraphLinear(hidden_dim * n_heads,
+                              n_edge_types * hidden_dim * n_heads)
+                  if i > 0 and concat_heads else
```

Comment:
Just a matter of preference, but how about setting
`in_channels` variable before hand with  `if i > 0 and concat_heads else` condition?